### PR TITLE
Dropzone

### PR DESCRIPTION
**Pull request by ConalCosgrove**
drop file capability

This Pull Request Adds the Ability to Upload Files to a Salesforce case connected to an IMC room via the Salesforce Case Bot.

### .eslintrc

- Turned off curly rule, allows one line if() 
- Turned off no magic numbers 
- Turned off func-style. 

### config.js

- Added parameters for salesforce org credentials 
- Added a toggle to enable/disable file upload 

### index.js

- Added a 'handleActions' function to receive uploaded files and send them to salesforce.

### package.json

- Updated babel and webpack as well as other packages.
- Added a botAction reference for "postAttachmentTimeline"

### salesforce.js

- Created this file to encapsulate the process of querying the Salesforce case and then attaching the file

### web/components/App.jsx

- Updated imports for react components
- Wrapped the timeline in a div which acts as the 'dropzone', added style objects to show a blue border when hovering over with a file. 
- Changed 'getDerivedStateFromProps' to 'componentDidUpdate'
- Added file upload logic for drag and drop
- Provided file upload function to `ChatBox` component for use in Attachment Button 

### web/components/ChatBox.js

- Refactored to remove unnecessary use of state. 
- Added 'Attach File' button which appears if upload feature is enabled

### web/components/dropzoneActions.js

- File that contains the logic for the dropzone component; handlers for fileDraggedOver, fileDraggedAway, and uploadFile

### web/components/postAttachment.js

- File that handles creating the botAction to send the uploaded file to the backend 

### web/index.js

- Refactoring
- Added code to parse the selected chatter of the room

### webpack.config.js

- Updated webpack config to deal with newer versions of webpack and babel

### CODE REVIEW

**Comment by diogo-pessoa** [comment]
Just have to fix the test failing in travis.

File: salesforce.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  userName,
+  caseNumber,
+  id,
+  selectedChatter
+) {
+  const chatterIndex = chatterOptionsArray.indexOf(selectedChatter);
+
+  // eslint-disable-next-line
+  if (!timelineBotAttachment) return;
+  if (chatterIndex === NOT_FOUND) {
```
Comment (LongweiDeng) [inline salesforce.js:87]: Does the function end when chatter index not found? 
Should it return or throw error?

Comment (ConalCosgrove) [reply salesforce.js:87]: yeah I suppose I could just include that check in the above if statement

**Comment by ConalCosgrove** [comment]
@here the integration is failing because we have disabled access to the staging version of heroku. All the unit tests are passing though

**Review by LongweiDeng — APPROVED** [review]

**Comment by LongweiDeng** [comment]
Everything else looks good to me

**Review by BrandonMurnane — APPROVED** [review]

**Pull request merged**